pkg/slareport/prom_test.go

```diff
@@ -0,0 +1,203 @@
+			startDate, endDate, err := parseRange(tt.args.date, tt.args.duration)
+			if tt.wantErr {
+				assert.Error(t, err)
+			} else {
+				assert.Equal(t, startDate, &tt.startDate)
+				assert.Equal(t, endDate, &tt.endDate)
+			}
+		})
+	}
+}
+
+func TestRunQuery(t *testing.T) {
+	tests := []struct {
+		name    string
+		metrics model.Matrix
+		want    map[string][]ServiceInstance
+		wantErr bool
+	}{
+		{
+			name: "GivenAvailableMetrics_ThenExpectOutput",
+			want: map[string][]ServiceInstance{
+				"mycustomer": {
+					{
+						Namespace:  "myns",
+						Instance:   "test",
+						TargetSLA:  99.9,
+						OutcomeSLA: 99.99,
+						Cluster:    "mycluster",
+						Service:    "postgresql",
+						Color:      "green",
+					},
+				},
+			},
+			metrics: model.Matrix{
+				{
+					Metric: model.Metric{
+						"name":         "test",
+						"namespace":    "myns",
+						"service":      "postgresql",
+						"cluster_id":   "mycluster",
+						"organization": "mycustomer",
+					},
+					Values: []model.SamplePair{
+						{
+							Timestamp: model.Time(time.Now().UnixMilli()),
+							Value:     model.SampleValue(99.99),
+						},
+						{
+							Timestamp: model.Time(time.Now().UnixMilli()),
+							Value:     model.SampleValue(99.99),
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "GivenUnreachedSLA_ThenExpectColorRed",
+			want: map[string][]ServiceInstance{
+				"mycustomer": {
+					{
+						Namespace:  "myns",
+						Instance:   "test",
+						TargetSLA:  99.9,
+						OutcomeSLA: 99.8,
+						Cluster:    "mycluster",
+						Service:    "postgresql",
+						Color:      "red",
+					},
+				},
+			},
+			metrics: model.Matrix{
+				{
+					Metric: model.Metric{
+						"name":         "test",
+						"namespace":    "myns",
+						"service":      "postgresql",
+						"cluster_id":   "mycluster",
+						"organization": "mycustomer",
+					},
+					Values: []model.SamplePair{
+						{
+							Timestamp: model.Time(time.Now().UnixMilli()),
+							Value:     model.SampleValue(99.8),
+						},
+						{
+							Timestamp: model.Time(time.Now().UnixMilli()),
+							Value:     model.SampleValue(99.8),
+						},
+					},
+				},
+			},
+		},
+		{
+			name:    "GivenNoMetrics_ThenExpectEmptyOutput",
+			want:    map[string][]ServiceInstance{},
+			metrics: model.Matrix{},
+		},
+		{
+			name: "GivenNoValues_ThenExpectAnError",
+			want: map[string][]ServiceInstance{},
+			metrics: model.Matrix{
+				{
+					Metric: model.Metric{
+						"name":         "test",
+						"namespace":    "myns",
+						"service":      "postgresql",
+						"cluster_id":   "mycluster",
+						"organization": "mycustomer",
+					},
+				},
+			},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+
+		promClientFunc = getDummyPromClient
+		getTargetSLAFunc = getDummySLA
+		getMetricsFunc = func(ctx context.Context, startDate, endDate *time.Time, timeRange string, client apiv1.API) (model.Matrix, error) {
+			return tt.metrics, nil
+		}
```

Comment:
Nitpick: You change global state in the test without reverting it. So this could potentially lead to weird behavior if someone adds an integration test and is confused why the actual `promClientFunc` isn't called